File: app/src/main/java/com/borisruzanov/russianwives/mvp/model/interactor/ActionsInteractor.java
package com.borisruzanov.russianwives.mvp.model.interactor;

import com.borisruzanov.russianwives.mvp.model.repository.FirebaseRepository;
import com.borisruzanov.russianwives.utils.ActionItemCallback;

public class ActionsInteractor {

    private FirebaseRepository repository;

    public ActionsInteractor(FirebaseRepository repository) {
        this.repository = repository;
    }

    public void getActions(ActionItemCallback callback){
        repository.getTransformedActions(callback);
    }

}

File: app/src/main/java/com/borisruzanov/russianwives/mvp/model/interactor/actions/ActionsInteractor.kt
package com.borisruzanov.russianwives.mvp.model.interactor.actions

import com.borisruzanov.russianwives.mvp.model.repository.user.UserRepository
import com.borisruzanov.russianwives.utils.ActionItemCallback
import javax.inject.Inject

class ActionsInteractor @Inject constructor(private val repository: UserRepository) {

    fun getActions(callback: ActionItemCallback) {
        repository.getTransformedActions(callback)
    }

}
